tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -113,34 +128,38 @@ func getLabels(record events.S3EventRecord) (map[string]string, error) {
 	return labels, nil
 }
 
-func processS3Event(ctx context.Context, ev *events.S3Event) error {
-	batch, err := newBatch(ctx)
+func parseS3Record(ctx context.Context, record events.S3EventRecord, batch batchIf) (bool, error) {
+	labels, err := getLabels(record)
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	for _, record := range ev.Records {
-		labels, err := getLabels(record)
-		if err != nil {
-			return err
-		}
+	obj, err := getS3Object(ctx, labels)
+	if err != nil {
+		return false, err
+	}
 
-		obj, err := getS3Object(ctx, labels)
-		if err != nil {
-			return err
-		}
+	err = parseS3Log(ctx, batch, labels, obj)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
 
-		err = parseS3Log(ctx, batch, labels, obj)
-		if err != nil {
-			return err
-		}
+func processS3Event(ctx context.Context, ev *events.S3Event) (bool, error) {
```

Comment:
What is the reason for changing the return value from `error` to `(bool, error)`?
First, there is no combination where `error` is not `nil` but the `bool` is true. And second, the `bool` return value is discarded in the only usage of `processS3Event()` (main.go L148)